packages/zone/src/heap.js

```diff
@@ -1,28 +1,25 @@
 // @ts-check
 // @jessie-check
 
+import { isPassable } from '@agoric/internal';
+import { makeExo, defineExoClass, defineExoClassKit } from '@endo/exo';
 import {
-  makeExo,
-  defineExoClass,
-  defineExoClassKit,
   makeScalarMapStore,
   makeScalarSetStore,
   makeScalarWeakMapStore,
   makeScalarWeakSetStore,
   M,
 } from '@agoric/store';
 
-import { Far } from '@endo/far';
-
 import { makeOnceKit } from './make-once.js';
 import { agoricVatDataKeys as keys } from './keys.js';
 
 /**
  * @type {import('.').Stores}
  */
-const detachedHeapStores = Far('heapStores', {
+const detachedHeapStores = harden({
   detached: () => detachedHeapStores,
-  isStorable: _specimen => true,
+  isStorable: isPassable,
```

Comment:
In an unpublished draft I tried reproducing in heap.js the `isStorable` check that #7891 has for virtual isStorable, but using a heap store rather than a virtual store. As I expected, the heap store refuses to store non-passables. Thus, I think the heap zone should consider non-passables to not be heap storable. 

As explained at https://github.com/endojs/endo/issues/1648 , heap exos will currently store non-passables. But I filed the issue because I consider that a bug. Your PR should proceed on the heap store precedent, assuming this bug is eventually fixed.